packages/core/src/keyBinding/keyBinding.ts

```diff
@@ -95,7 +102,15 @@ export class KeyBinding<
         requiredClaims?: Array<keyof Payload | string>,
         publicKeyJwk?: Record<string, unknown>
     ): Promise<KeyBindingVerificationResult> {
-        this.assertValidForKeyBinding()
+        if (!this.expectedSdHash) {
+            throw new Error(
+                'Expected sd hash is required for verification of key binding JWT'
+            )
+        }
+
+        // TODO: should _sd_hash also be a verification property (true/false)
+        // or should it throw?
+        await this.assertValidForKeyBinding(this.expectedSdHash)
```

Comment:
We can leave it to throw for now, but I think in the future a verification property would be more inline with the rest of the API.

Reply:
Ah yeah i was really doubting as the other checks (like whether `iat` is present) is not part of the boolean result values but will throw an error directly. Should we also move these to the validation object?

I think it's nice if it will always return a result object rather than sometimes throwing